tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -4,47 +4,133 @@
 // </copyright>
 
 using System;
+using System.IO;
 using System.Net;
+using System.Text;
 
-using Datadog.Trace.Logging;
-using Datadog.Trace.Util;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
 
 namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS
 {
     internal class LambdaCommon
     {
-        private const string TraceContextEndpointEnvName = "_DD_TRACE_CONTEXT_ENDPOINT";
-        private const string TraceContextEndpoint = "http://127.0.0.1:8124/lambda/trace-context";
         private const string PlaceholderServiceName = "placeholder-service";
         private const string PlaceholderOperationName = "placeholder-operation";
+        private static readonly string DefaultJson = "{}";
 
-        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LambdaCommon));
+        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder)
+        {
+            var json = DefaultJson;
+            try
+            {
+                json = JsonConvert.SerializeObject(payload);
+            }
+            catch (Exception)
+            {
+                Serverless.Debug("Could not serialize input");
+            }
+
+            NotifyExtensionStart(requestBuilder, json);
+            return new CallTargetState(CreatePlaceholderScope(Tracer.Instance, requestBuilder));
+        }
+
+        internal static CallTargetState StartInvocationWithoutEvent(ILambdaExtensionRequest requestBuilder)
+        {
+            return StartInvocation(DefaultJson, requestBuilder);
+        }
+
+        internal static CallTargetReturn<TReturn> EndInvocationSync<TReturn>(TReturn returnValue, Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
+        {
+            NotifyExtensionEnd(requestBuilder, exception != null);
+            scope?.ServerlessDispose();
```

Comment:
If you move `ServerlessDispose()` into this file (instead of making it a global extension method, see my other comment), you'll need to change this syntax to:
```csharp
ServerlessDispose(span);
```

Reply:
✅